test/rekt/features/broker/data_plane.go

```diff
@@ -407,3 +414,440 @@ func brokerEventVersionNotUpgraded(ctx context.Context, t feature.T) {
 
 	// Create a trigger,
 }
+
+func deliveredCESpecs(ctx context.Context, t feature.T) {
+	brokerName := state.GetStringOrFail(ctx, t, "brokerName")
+
+	uuids := map[string]string{
+		uuid.New().String(): "1.0",
+		uuid.New().String(): "0.3",
+	}
+
+	for id, version := range uuids {
+		source := feature.MakeRandomK8sName("source")
+		sink := feature.MakeRandomK8sName("sink")
+		triggerName := feature.MakeRandomK8sName("trigger")
+		event := test.FullEvent()
+		event.SetID(id)
+		event.SetSpecVersion(version)
+
+		// Creating sink
+		eventshub.Install(sink, eventshub.StartReceiver)(ctx, t)
+
+		// Point the Trigger subscriber to the sink svc.
+		cfg := []manifest.CfgFn{trigger.WithSubscriber(service.AsKReference(sink), "")}
+
+		// Creating trigger
+		trigger.Install(triggerName, brokerName, cfg...)
+
+		eventshub.Install(source,
+			eventshub.StartSenderToResource(broker.GVR(), brokerName),
+			eventshub.InputEvent(event),
+		)
+
+		assert.OnStore(sink).MatchEvent(test.HasId(event.ID()), test.HasSpecVersion(version)).Exact(1)
+	}
+}
```

Comment:
I do not fully get this test here.

The text says `Delivered events MUST conform to the CloudEvents specification."`

Instead of sending two valid CEs and checking for the presence of the version (why should it not be there?), I'd suggest to send a couple of events. valid and invalid ones (e.g. missing the `source` attribute), and make sure we really receive only the valid ones.